src/cmake-tools.ts

```diff
@@ -2138,14 +2136,37 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
         return vscode.debug.activeDebugSession!;
     }
 
-    private _launchTerminal?: vscode.Terminal;
+    private _launchTerminals = new Map<number, vscode.Terminal>();
+    private _launchTerminalTargetName = '_CMAKE_TOOLS_LAUNCH_TERMINAL_TARGET_NAME';
     // Watch for the user closing our terminal
-    private readonly _termCloseSub = vscode.window.onDidCloseTerminal(term => {
-        if (term === this._launchTerminal) {
-            this._launchTerminal = undefined;
+    private readonly _termCloseSub = vscode.window.onDidCloseTerminal(async term => {
+        const processId = await term.processId;
+        if (this._launchTerminals.has(processId!)) {
+            this._launchTerminals.delete(processId!);
         }
     });
 
+    private _createTerminal(options: vscode.TerminalOptions, executable: api.ExecutableTarget) {
+        if (!this.workspaceContext.config.launch_allowParallel) {
+            for (const [, term] of this._launchTerminals) {
+                const creationOptions = term.creationOptions! as vscode.TerminalOptions;
+                const executablePath = creationOptions.env![this._launchTerminalTargetName];
+                if (executablePath === executable.name) {
+                    if (this.workspaceContext.config.launch_terminatePreviousInstance) {
+                        term.sendText('\u0003');
+                    }
+                    return term;
+                }
+            }
+        }
+
+        if (options && options.env) {
+            options.env[this._launchTerminalTargetName] = executable.name;
+        }
```

Comment:
This was missing from the original implementation. The above check for `creationOptions.env![this._launchTerminalTargetName]` wasn't working because the env var wasn't set to begin with, causing a previous instance to not be terminated when settings require.